### ISSUE TITLE
Z-Code++: A Pre-Trained Language Model Optimized for Abstractive Summarization

### ISSUE THREAD

**Issue by ichiroex** [ichiroex/nlp-papers#17]
> He, Pengcheng, Baolin Peng, Liyang Lu, Song Wang, Jie Mei, Yang Liu, Ruochen Xu, et al. 2022. “Z-Code++: A Pre-Trained Language Model Optimized for Abstractive Summarization.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/2208.09770.

## 概要
- 抽象型要約向けに最適化された新たな事前学習済み言語モデルの提案
  - 関連モデル -> PUGASUS, T5
- 主に3つのテクニックを導入
  - (1) 低資源の要約問題を改善するために`two-phase pretraining process`による学習
  - (2) 通常のself-attentionの代替として， `disentangled attention` を導入
  - (3) long sequenceに対応するために，hierarchicalなエンコード手法 `fusion-in-encoder`の導入

- What's Z-Code
  - https://www.microsoft.com/en-us/research/project/project-zcode/

### 3.1 Two-phase pretraining process

- Two-phase pretraining process: language understanding -> grounded text generation 
- Two pre-training tasks for language understanding
  - replaced token detection（RTD）(Clark et al., 2020) ※ ELECTRA論文
    - generatorはMLMで学習
  - corrupted span prediction（CSP）(Raffel et al., 2020) ※ T5論文
    - spanはランダムに選択する. 平均長は3 tokens
- RTDで学習する時の工夫
  - ELECTRAでは，discriminatorとgeneratorでtoken embeddingsを共有している
  - MLMは意味的に類似したembedingを近づけるが、RTDでは意味的に近いembeddingを識別して遠ざけることをしており、効率がよくない
  - そこで本研究では、（He et al., 2021）の手法に従って，discrimatorのtoken embeddingsをre-paramerizeする
  - ここで`sg` は stop gradient operatorを表す
  - discriminatorのtoken embeddings を更新する際に $\boldsymbol{E}_{\Delta}$のみをupdate

<img width="936" alt="スクリーンショット 2022-09-27 12 12 37" src="https://user-images.githubusercontent.com/13919583/192423278-aa779ab8-939b-49d9-ab60-4fa9a6a8c898.png">

<img width="941" alt="スクリーンショット 2022-09-27 11 19 59" src="https://user-images.githubusercontent.com/13919583/192416293-3ce98c7b-cd57-47f0-8b83-7b13f49284e3.png">

low-resource summarizationに対して、grounded pre-traiingを適用
<img width="895" alt="スクリーンショット 2022-09-27 11 20 38" src="https://user-images.githubusercontent.com/13919583/192416376-88171535-d456-42af-acb6-ceb52b546eba.png">

- Second phaseでは、以下のデータセットを用いて grounded pre-training を実施した
- Instructionの例
<img width="913" alt="スクリーンショット 2022-09-27 11 22 08" src="https://user-images.githubusercontent.com/13919583/192416577-2a29ea01-6f72-46b8-8921-6aa5542e6840.png">

### 3.2 Disentangled attention (He et al., 2020; 2021)
- 単語をcontent embeddingとposition embeddingの2つのベクトルにより表現する方法

### 3.3 Fusion-in-Encoder（FiE）
- input sequenceをchunk単位に分割し、chunkごとにlocalなattentionを適用してchunk representationを作成
- その後、chunk単位による通常のself-attentionを適用
- hierarchicalな構成にすることでlong sequenceを扱えるようになった
- 実験では input lengthが10k words以上のとき、FiEを導入する（arXiv, PubMed 等）

<img width="716" alt="スクリーンショット 2022-09-27 11 17 41" src="https://user-images.githubusercontent.com/13919583/192415997-f234144c-9e4f-4aa6-ae05-368eade84be2.png">

## 結果

- データセット 
  - standard
  - long
  - multilingual
  - Z-Code++ largeは, 160GBのデータ、128kのvocab size
    - 160G English text data と section1で言及しているが、具体的にどのデータか記載されていない（社内データ？）
  - multilingualの設定では、mC4 data, 250kのvocab size -> mT5と同じ
    - 計算リソースの制約により、500B tokensのみ学習。（ただし、mT5では1T tokens学習されている）
- ハイパラ
  - grid searchで決めた. 詳細はAppendix 1. 
  - 1M steps学習、batch sizeは2048
  - AdamW is used as the optimizer

- 評価用データセット
<img width="955" alt="スクリーンショット 2022-09-27 14 00 36" src="https://user-images.githubusercontent.com/13919583/192435991-d0a9e8d1-c456-42a6-9d9b-6ab770192a12.png">

- Common English Summarization tasks (Standard）
  - PEGASUSを上回る性能
  - PEGASUSはニュースデータで学習されているので、Xsum, CNNDMではそこまで差がない
  - 一方で、Z-Code++ は多様なweb dataで学習されているので、一般ドメインにより適用しやすい
<img width="967" alt="スクリーンショット 2022-09-27 14 31 29" src="https://user-images.githubusercontent.com/13919583/192440248-9d002e9b-e610-4498-b478-d4287d43e1b5.png">

- Long Document Summarization
  - long document summarizationに最適化されたlongT5を上回る性能を達成
  - LongT5 xlargeに比べて、パラメータ数は 1/3だが同等の性能を達成
  - （この結果だけでは、disentangled attentionの有用性はわからない気が...）
<img width="951" alt="スクリーンショット 2022-09-27 14 32 48" src="https://user-images.githubusercontent.com/13919583/192440439-483f1491-fc3c-43a5-8b26-5035ecf91c96.png">

- Human evaluation（評価セットは、XSum）
  - side-by-sideで評価
  - hallucinationの面で改善が見受けられる
  - https://leaderboard.allenai.org/genie-xsum/submissions/public
<img width="877" alt="スクリーンショット 2022-09-27 14 42 34" src="https://user-images.githubusercontent.com/13919583/192441881-347b4bd9-5902-4e70-af05-163c62ab1a8f.png">

- multilingual summarization
  - パラメータ数、データ量がより大きい mT5 xlargeを上回っている
<img width="1002" alt="スクリーンショット 2022-09-27 14 46 30" src="https://user-images.githubusercontent.com/13919583/192442937-3869fac4-0c41-42e0-9bc7-811ba1024c35.png">

- low-resource summarization
  - first phaseだけでは、gap sentence generationで学習するPEGASUSには勝てていない
  - second phaseを導入することで、zero-shot, few-shotなどでPEGASUSを上回る性能を達成した
<img width="963" alt="スクリーンショット 2022-09-27 14 53 29" src="https://user-images.githubusercontent.com/13919583/192443967-4a4a8944-173e-4933-8858-f90a4cae2f40.png">

## 議論
- 160GB English Textとは
- 各テクニックの有用性は？
  - どのように評価すればよいか
- 大規模言語モデルの研究の場合、網羅的なablation studyをする際には潤沢な計算リソースが必要？
- Table 2の95%とは？（長さ上位95%でどのくらいか？）
- たしかに実験結果ではROUGEが上回っているが、意味のある差なのか
  - ハイパラ探索を頑張っている印象があり、実際はどうなのか。
- 人手評価が大事とはいえ、結局１つのデータでしか検証していない

## 参考にしたいところ
- ROUGE-2 Fのみ報告。その他の詳細はappendix。
 - recallなのかf-measureなのか明記していない論文もある
 - 詳細をappendixに回すことでスペースを節約
 - 各モデルのパラメータ数を明記
 

## 参考
- [Z-code project by Microsoft](https://www.microsoft.com/en-us/research/project/project-zcode/)
